### PR TITLE
point to xcode 12 beta1

### PR DESCRIPTION
**Pull request by kaisong1990**
## Proposed changes

First draft against xcode 12 beta1 build, will keep this branch here till all issues are addressed.

For any issues we found(any functionality or testing) from both iOS 14 and earlier version, we will make a fix on our dev branch. I will rebase this branch on top of our dev branch weekly to make sure everything is also running find in Xcode 12

## Type of change

- [ ] Feature work
- [ ] Bug fix
- [ ] Documentation
- [ ] Engineering change
- [x] Test
- [ ] Logging/Telemetry

## Risk

- [ ] High – Errors could cause MAJOR regression of many scenarios. (Example: new large features or high level infrastructure changes)
- [ ] Medium – Errors could cause regression of 1 or more scenarios. (Example: somewhat complex bug fixes, small new features)
- [x] Small – No issues are expected. (Example: Very small bug fixes, string changes, or configuration settings changes)

### CODE REVIEW

**Comment by kaisong1990** [comment]
The latest commit on fixing keychain accessing issue is also applicable on macOS 10.15+

File: MSAL/test/unit/utils/MSALTestCacheTokenResponse.m [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+                                                              response:msidResponse
+                                                               factory:[MSIDAADV2Oauth2Factory new]
+                                                               context:nil
+                                                                 error:error];
+}
+
++ (MSIDAADV2TokenResponse *)msalDefaultTokenResponseWithAuthority:(NSString *)authorityString
+                                                         familyId:(NSString *)familyId
+{
+    NSDictionary* idTokenClaims = @{ @"home_oid" : @"myuid", @"preferred_username": @"fakeuser@contoso.com", @"tid": @"utid"};
```
Comment (oldalton) [inline MSAL/test/unit/utils/MSALTestCacheTokenResponse.m:58]: code style for star position

File: MSAL/test/unit/utils/MSALTestCacheTokenResponse.m [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+    MSIDConfiguration *configuration = [MSALTestCacheTokenResponse msalDefaultConfigurationWithAuthority:authorityString];
+    
+    return [tokenCacheAccessor saveTokensWithConfiguration:configuration
+                                                              response:msidResponse
+                                                               factory:[MSIDAADV2Oauth2Factory new]
+                                                               context:nil
+                                                                 error:error];
+}
+
++ (MSIDAADV2TokenResponse *)msalDefaultTokenResponseWithAuthority:(NSString *)authorityString
```
Comment (oldalton) [inline MSAL/test/unit/utils/MSALTestCacheTokenResponse.m:55]: Can you double check if we already have similar utilities in common repository?

File: MSAL/test/unit/utils/MSALTestCacheTokenResponse.h [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+#import <Foundation/Foundation.h>
+
+@class MSIDDefaultTokenCacheAccessor;
+@class MSIDAADV2TokenResponse;
+@class MSIDConfiguration;
+
+@interface MSALTestCacheTokenResponse : NSObject
+
++(BOOL)msalStoreTokenResponseInCacheWithAuthority:(NSString *)authorityString
```
Comment (oldalton) [inline MSAL/test/unit/utils/MSALTestCacheTokenResponse.h:36]: nit: add a space after +

File: MSAL/test/unit/utils/MSALTestCacheTokenResponse.m (hunk 1 above)
Comment (kaisong1990) [reply MSAL/test/unit/utils/MSALTestCacheTokenResponse.m]: Updated

File: MSAL/test/unit/utils/MSALTestCacheTokenResponse.h (hunk 3 above)
Comment (kaisong1990) [reply MSAL/test/unit/utils/MSALTestCacheTokenResponse.h]: Updated

File: MSAL/test/unit/utils/MSALTestCacheTokenResponse.m (hunk 2 above)
Comment (kaisong1990) [reply MSAL/test/unit/utils/MSALTestCacheTokenResponse.m]: Refactored this method by removing authority string and reused method from identity core to create token response. Still need to keep the method as a wrapper for some customization/setting ups

**Pull request merged**